src/ngraph/frontend/onnx_import/op/lstm.cpp

```diff
@@ -0,0 +1,371 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+#include <iterator>
+#include <map>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "ngraph/node.hpp"
+#include "ngraph/op/add.hpp"
+#include "ngraph/op/concat.hpp"
+#include "ngraph/op/dot.hpp"
+#include "ngraph/op/multiply.hpp"
+#include "ngraph/op/reshape.hpp"
+#include "ngraph/op/sigmoid.hpp"
+#include "ngraph/op/tanh.hpp"
+#include "ngraph/shape.hpp"
+#include "ngraph/type/element_type.hpp"
+
+#include "exceptions.hpp"
+#include "lstm.hpp"
+#include "utils/broadcasting.hpp"
+#include "utils/common.hpp"
+#include "utils/reshape.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace
+            {
+                std::shared_ptr<ngraph::Node> add(const std::shared_ptr<ngraph::Node>& lhs,
+                                                  const std::shared_ptr<ngraph::Node>& rhs)
+                {
+                    auto args = numpy_style_broadcast_for_binary_operation(lhs, rhs);
+                    return {std::make_shared<ngraph::op::Add>(args.at(0), args.at(1))};
+                }
+
+                std::shared_ptr<ngraph::Node> mul(const std::shared_ptr<ngraph::Node>& lhs,
+                                                  const std::shared_ptr<ngraph::Node>& rhs)
+                {
+                    auto args = numpy_style_broadcast_for_binary_operation(lhs, rhs);
+                    return {std::make_shared<ngraph::op::Multiply>(args.at(0), args.at(1))};
+                }
+
+                using NgraphNodePtr = std::shared_ptr<ngraph::Node>;
+
+                // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ACTIVATION FUNCTIONS ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+                NgraphNodePtr Sigmoid(const NgraphNodePtr& arg)
+                {
+                    return std::make_shared<ngraph::op::Sigmoid>(arg);
+                }
+
+                NgraphNodePtr Tanh(const NgraphNodePtr& arg)
+                {
+                    return std::make_shared<ngraph::op::Tanh>(arg);
+                }
+
+                // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~ INPUT NODES PARSING ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+                struct LSTMNgInputMap
+                {
+                    using iterator = std::map<std::string, NgraphNodePtr>::iterator;
```

Comment:
```
using container_type = std::map<std::string, std::shared_ptr<ngraph::Node>>;
using iterator = typename container_type::iterator;
...
```

Reply:
The convention `aaa_bbb` is the standard library convention for class traits. Standard algorithms use them a lot.